src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewAddColumnDialog.cs

```diff
@@ -0,0 +1,1005 @@
+
+        host = _liveDataGridView?.Site?.GetService(iDesignerHostType) as IDesignerHost;
+        if (host is not null)
+        {
+            container = host.Container;
+        }
+
+        while (!ValidName(columnName,
+                            _dataGridViewColumns!,
+                            container,
+                            null /*nameCreationService*/,
+                            _liveDataGridView!.Columns,
+                            !_persistChangesToDesigner))
+        {
+            colId++;
+            columnName = "Column" + colId.ToString(CultureInfo.InvariantCulture);
+        }
+
+        return columnName;
+    }
+
+    /// <summary>
+    ///  Clean up any resources being used.
+    /// </summary>
+    protected override void Dispose(bool disposing)
+    {
+        base.Dispose(disposing);
+    }
+
+    private void EnableDataBoundSection()
+    {
+        bool dataGridViewIsDataBound = _dataColumns!.Items.Count > 0;
+        if (dataGridViewIsDataBound)
+        {
+            _dataBoundColumnRadioButton!.Enabled = true;
+            _dataBoundColumnRadioButton.Checked = true;
+            _dataBoundColumnRadioButton.Focus();
+            _headerTextBox!.Text = _nameTextBox!.Text = AssignName();
+        }
+        else
+        {
+            _dataBoundColumnRadioButton!.Enabled = false;
+            _unboundColumnRadioButton!.Checked = true;
+            _nameTextBox!.Focus();
+            _headerTextBox!.Text = _nameTextBox.Text = AssignName();
+        }
+    }
+
+    /// <summary>
+    ///  public because the EditColumns dialog wants to use it as well.
+    /// </summary>
+    /// <param name="typeResolutionService"></param>
+    /// <param name="type"></param>
+    /// <returns></returns>
+    public static ComponentDesigner? GetComponentDesignerForType(ITypeResolutionService? typeResolutionService, Type type)
+    {
+        ComponentDesigner? compDesigner = null;
+        DesignerAttribute? designerAttribute = null;
+        AttributeCollection? attributes = TypeDescriptor.GetAttributes(type);
+        for (int i = 0; i < attributes.Count; i++)
+        {
+            DesignerAttribute? attribute = attributes[i] as DesignerAttribute;
+            if (attribute is not null)
+            {
+                Type? daType = Type.GetType(attribute.DesignerBaseTypeName);
+                if (daType is not null && daType == iDesignerType)
+                {
+                    designerAttribute = attribute;
+                    break;
+                }
+            }
+        }
+
+        if (designerAttribute is not null)
+        {
+            Type? designerType = null;
+            if (typeResolutionService is not null)
+            {
+                designerType = typeResolutionService.GetType(designerAttribute.DesignerTypeName);
+            }
+            else
+            {
+                designerType = Type.GetType(designerAttribute.DesignerTypeName);
+            }
```

Comment:
This can be simplified.
```suggestion
            Type? designerType = typeResolutionService is null 
            ? Type.GetType(designerAttribute.DesignerTypeName) 
            : typeResolutionService.GetType(designerAttribute.DesignerTypeName);
```